### ISSUE TITLE
feat: add `cloudquery validate` subcommand to validate a configuration directory

### ISSUE THREAD

**Issue by asaba-hashi** [cloudquery/cloudquery#15704]
### Which problem is this feature request solving?

I would like to check that my configuration is syntactically valid and internally consistent without invoking a full `sync` operation or making any actual API requests using real credentials.

For my use case, the following errors would be nice to catch in CI:

1. Syntactically correct yaml
1. Invalid plugin versions
1. Mismatched plugin versions
1. Missing required fields
1. `spec` should also be validated for syntactic correctness and internal consistency with it's plugin provider.
1. etc

Some of these can be checked with variations and combinations of other commands, but the ask is for:

1. A single subcommand.
2. No side-effects (such as plugin downloads or source/destination API calls).
3. List all problems found instead of failing at the first one.

### Describe the solution you'd like

Add a `cloudquery validate` subcommand that takes similar arguments to `sync`, that just does the minimal loading a checks of the configuration.  If necessary, the user should supply dummy variables values for testing.  Given the following file:

```yaml
kind: source
spec:
  name: aws
  path: cloudquery/aws
  registry: cloudquery
  version: "v19.09.2"
  spec:
    accounts:
       - id: 123456789012
         role: {ROLE_ARN}
         role_session_name: CloudQuerySession
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
```

`cloudquery validate .` should fail with a message listing all detectable errors (i.e. not stopping after the first one).

```
ROLE_ARN is not defined.

plugin version is invalid for aws
```

These checks would be similar to something like what [nomad job validate](https://developer.hashicorp.com/nomad/docs/commands/job/validate) is able to check.

The command should not make any changes or fetch new versions of plugins.

### Pull request (optional)

- [ ] I can submit a pull request

**Comment by erezrokah** [comment]
Hi @asaba-hashi, thanks for opening the issue.

> 1. Invalid plugin versions

You should be able to achieve this by running `cloudquery plugin install` (see https://docs.cloudquery.io/docs/reference/cli/cloudquery_plugin_install). The command will fail if a plugin version is incorrect, or download all plugins if all versions are correct. Downloaded plugins are cached for the next run

> 2. Mismatched plugin versions

You should be able to achieve this by running `cloudquery migrate` (see https://docs.cloudquery.io/docs/reference/cli/cloudquery_migrate). The command verifies the versions are compatible and then migrates any relevant tables.
It should work with dummy variables for sources, but would still require a valid connection to the destinations to perform the migration

**Comment by asaba-hashi** [comment]
Thanks @erezrokah.

To clarify, those where 2 examples from a non-exhaustive list. The feature request is still for a single, consistent command that specifically does _not_ take _any_ action like the `migrate` described above.

**Comment by erezrokah** [comment]
Thanks for the added context @asaba-hashi, do you mind sharing the other use cases from the list? That should help us verify we address all the needed requirements

**Comment by asaba-hashi** [comment]
> Thanks for the added context @asaba-hashi, do you mind sharing the other use cases from the list? That should help us verify we address all the needed requirements

@erezrokah I have updated the description with some additional errors that could have been caught earlier with no additional configuration or interaction with other systems. I updated the "like this" reference to `nomad job validate`, since cloudquery configurations are much more like nomad jobs.

**Comment by erezrokah** [comment]
Hi 👋 

In CLI version `v5.10.0` we've added an experimental `cloudquery test-connection`  that validates the spec and ensures plugins connection to related services work (e.g. AWS auth, Postgres DB connection)

Feedback welcomed

**Comment by myoung34** [comment]
Hey, this isnt quite what I wished.

this requires valid connections to cloud(s), backends, etc. I simply want to do a schema validate without making our CI have valid credentials to the data sources (espcially to the output destinations)

```
✗ mkdir config
✗ cat configmap.yaml| yq .data | grep -v ^'config:' | sed 's/^  //g' | yq >config/config.yaml
✗ docker run -v $(pwd)/config:/config ghcr.io/cloudquery/cloudquery:latest test-connection /config
Loading spec(s) from /config
Error: unauthorized. Try logging in via `cloudquery login`
``` 

I dont wish to log in to do a schema validation

**Comment by erezrokah** [comment]
Hi @myoung34 we have ongoing work that should satisfy your requirement I believe https://github.com/cloudquery/cloudquery/pull/17530

**Comment by bbernays** [comment]
We have added a cloudquery `validate-config` command. This new command will still require the plugin to be downloaded and available locally, but it will not require any credentials for the source/destination ie (AWS/GCP/Postgres). A few caveats to keep in mind:
1. You will still have to authenticate with the CQ API to download the plugin (or have already downloaded the plugin and set the `cq-dir` )
2. This will do a stricter validation of the config then we currently require when running sync
3. This command will not validate that the tables are valid as that requires the plugin to be initialized

This functionality is available now in `v5.14.0`
